File: proj/CoreLibrary/src/com/jasonzqshen/familyaccounting/core/exception/FiscalYearRangeException.java
package com.jasonzqshen.familyaccounting.core.exception;

public class FiscalYearRangeException extends Exception {
	/**
	 * 
	 */
	private static final long serialVersionUID = 6650231174034410258L;

	
	public FiscalYearRangeException(int fiscalYear) {
		super(String.format(
				"Fiscal Year range is from 1000 to 9999. But the value is %d",
				fiscalYear));
	}
}

File: proj/WinStore_FamilyAccounting/WinStore_FamilyAccountingCore/Exceptions/FiscalYearRangeException.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WinStore_FamilyAccountingCore.Exceptions
{
    public class FiscalYearRangeException: Exception
    {
        public FiscalYearRangeException(int fiscalMonth)
            : base(String.Format("Fiscal Year range is from 1000 to 9999. But the value is %d",
                fiscalMonth))
        {
        }	
    }
}
